text/0000-fabric-private-chaincode-lite.md

```diff
@@ -240,129 +238,138 @@ The FPC-Lite architecture is constituted by a set of components which are design
+**(Note that FPC Lite makes the simplifying assumption that, for each deployed FPC Chaincode, a single enclave is initialized and registered.)**
 
 The administrator of the peer hosting the FPC chaincode enclave is responsible for the initialization and registration of the enclave.
-The operation is performed by executing the `initEnclave` admin command.
-
+The operation is performed by executing the `initEnclave` admin command (see step 2. above).
 The command can be triggered through the FPC Client SDK (see [earlier Deployment Section](#deployment)).
 <!--
 Alternatively, the administrator can use the Fabric client to:
 query the `initEnclave` FPC chaincode function;
 pass the result as an argument in the invocation to the `registerEnclave` Enclave Registry chaincode function.
 -->
-A successful initialization and registration will result in a new entry (the enclave credentials, which include the chaincode definition, attestation and key material) in the registry namespace on the ledger.
 
-Internally, the command operates as follows.
-First, it issues an `initEnclave` query which reaches the FPC Shim.
-The shim initializes the enclave (which will process FPC chaincode invocations) with the chaincode parameters received from the peer, namely: the chaincode definition and the channel identifier.
-Then, it generate the key material: public/private key-pairs for signing and encryption, and the state encryption key.
-The public signing key is used as enclave identifier.
-The shim completes the execution by returning the `credentials` of the FPC Chaincode: all public parameters and a hardware-based remote attestation covering them.
-Importantly, the code identity (or hash of the FPC Chaincode) specified in the attestation should match the version in the chaincode definition.
+A successful initialization and registration will result in a new entry in the enclave registry namespace on the ledger. In particular, each entry contains enclave credentials, which cryptographically bind the enclave to the chaincode as defined in the chaincode definition.
 
-In the case of EPID-based Intel® SGX, attestations must be converted into a publicly-verifiable evidence by contacting IAS.
-The FPC Client SDK performs this step using the admin's IAS subscription.
+Internally, the initialization command operates as follows:
 
+First, it issues an `initEnclave` query which reaches the FPC Shim.
+The shim initializes (creates) the enclave (which will process FPC chaincode invocations) with the chaincode parameters received from the peer, namely: the chaincode definition and the channel identifier.
+Then, the enclave generates public/private key-pairs for signing and encryption. The public signature key is used as a enclave identifier.  The key material for chaincode state encryption is generated in a subsequent key generation protocol and we refer to [FPC specification](../images/fpc/full-detail/fpc-key-dist.png) for more details.
```

Comment:
```suggestion
Then, the enclave generates public/private key-pairs for signing and encryption.
The public signature key is used as a enclave identifier. 
The key material for chaincode state encryption and the chaincode public encryption key are generated in a subsequent key generation protocol and we refer to [FPC specification](../images/fpc/full-detail/fpc-key-dist.png) for more details.
```

Reply:
> We also generate the chaincode encryption key pair.

We should discuss this @mbrandenburger @g2flyer , because the chaincode encryption key pair is related solely to the key distribution part.

This RFC is the authoritative document that will be examined.
It definitely has to describe the basic solution we want to transfer to Fabric.
Maybe it could also hint at solutions to future challenges (e.g., the trusted ledger, key distribution), although these are better postponed to future proposals.

The reviewers will look at the RFC, and not necessarily at the design documents.
The RFC therefore must be self-contained.
Any question must find an answer within the RFC.

This chaincode encryption key pair raises a question: "Why?".
This question does not find an answer in the RFC.
So this would impose us to write one, but we don't have it.
So we would put a reference to a future version of FPC.
In this case, the question is: "why are you writing about that at all?".

Assuming we do not write about it, RFC and design docs are slightly out-of-sync.
Is this a problem? No, because the RFC is authoritative for reviewers, not the design docs.
What about references to docs and reviewers seeing something different? This would be a terrific opportunity for an off-line discussion about key distribution, to show that we are actually ahead and staging work.

Concluding, there is no point in writing about that key pair in the RFC.
Chances are that we will be asked to remove it from the design,
and to add it later when we have the design and implementation of the key distribution part.

Reply:
(a bit too late for this PR, but what's matter is the discussion on the topic)